src/event.h

```diff
@@ -729,6 +730,15 @@ struct event_spec<event_type::uses_debug_menu> {
     };
 };
 
+template<>
+struct event_spec<event_type::u_var_changed> {
+    static constexpr std::array<std::pair<const char *, cata_variant_type>, 2> fields = { {
```

Comment:
Are `u_var`s associated with a particular character, or global? I guess the latter, but if the former, then the event should have a character id as well.

Reply:
At the point the event gets sent, the variable's associations are tracked by adding prefixes to their name. I created the `get_var_basename` function because any programmer has no way of finding that out. The downside is that we can't limit it to a certain character without changing that naming approach